server/mongodb/actions/ForgotPassword.js

```diff
@@ -0,0 +1,58 @@
+import bcrypt from "bcryptjs";
+import crypto from "crypto";
+import jwt from "jsonwebtoken";
+import mongoDB from "../index";
+import PasswordResetRequest from "../models/PasswordResetRequest";
+import User from "../models/User";
+import { sendEmail } from "../../actions/email.js";
+import urls from "../../../utils/urls.js";
+
+export async function forgotPassword(email) {
+  if (email == null) {
+    throw new Error("Email must be provided");
+  }
+
+  await mongoDB();
+
+  try {
+    const token = crypto.randomBytes(128).toString();
+    const hashedToken = await bcrypt.hash(token, 10);
```

Comment:
Originally didn't think hashing the tokens would be beneficial since the requests are relatively short-lived, but it doesn't hurt. Good catch :+1:

Reply:
Could you make this a function somewhere so that it can be re-used by the token look-up?

Reply:
Not sure what you mean by this - check my latest commit, I made some changes to the cryptographic part